### ISSUE TITLE
Missleading equation for ELP set

### ISSUE THREAD

**Issue by kacouane** [riscv/riscv-cfi#204]
Hello,

I've been discussing the spec and It seems that the ELP update equation is quite hard to follow

My understanding is three mechanisms can update the ELP {secured jumps, xret, landing pads} one is given in [spec section 3.1](https://raw.githubusercontent.com/riscv/riscv-cfi/main/riscv-cfi.pdf#%5B%7B%22num%22%3A103%2C%22gen%22%3A0%7D%2C%7B%22name%22%3A%22XYZ%22%7D%2C0%2C477.63%2Cnull%5D)

`ELP <- is_lp_expected`

the other is from [spec section 3.5](https://raw.githubusercontent.com/riscv/riscv-cfi/main/riscv-cfi.pdf#%5B%7B%22num%22%3A139%2C%22gen%22%3A0%7D%2C%7B%22name%22%3A%22XYZ%22%7D%2C0%2C841.89%2Cnull%5D)
and can be written something as:

`ELP <- is_xret_lp_expected = ( (MRET && MSTATUS.MPELP && getxLPE(MSTATUS.MPP) ) || ( SRET && SSTATUS.SPELP && getxLPE(SSTATUS.SPP) ) ) ? 1 : 0`

where `getxLPE(myPriv)` is tab 2 or 3 from [spec section 3.3](https://raw.githubusercontent.com/riscv/riscv-cfi/main/riscv-cfi.pdf#%5B%7B%22num%22%3A130%2C%22gen%22%3A0%7D%2C%7B%22name%22%3A%22XYZ%22%7D%2C0%2C544.1865%2Cnull%5D) but were the privilege mode is the `myPriv` variable 

And the last is in [spec section 3.4](https://raw.githubusercontent.com/riscv/riscv-cfi/main/riscv-cfi.pdf#%5B%7B%22num%22%3A135%2C%22gen%22%3A0%7D%2C%7B%22name%22%3A%22XYZ%22%7D%2C0%2C841.89%2Cnull%5D)

`ELP <- is_postjump_lp_expected = ( ( ELP == LP_EXPECTED ) && xLPE && ( pc[1:0] == 0 ) && ( ( instr.LPL == x7[31:12] ) || ( instr.LPL == 0 ) ) ? 0 : 1`

So the full equation would be something like this:

`ELP <- ( JALR || C.JR || C.JALR ) ? is_lp_expected : ( MRET || SRET ) ? is_xret_lp_expected : ( ( ELP == LP_EXPECTED ) && xLPE ) ? is_postjump_lp_expected : 0`

Am I correct?

**Comment by ved-rivos** [comment]
The equation seems right  (with `vsstatus` substituting for `sstatus` when `v=1`).
It is missing two cases from section 3.5:
1. DRET sets `ELP` to `dcsr.pelp`
2. Trap to VS-, HS/S-, M-, or D-mode saves `ELP` to `vsstatus.spelp`, `mstatus.mpelp` or `mstatus.spelp` or to `dcsr.pelp` respective and sets `ELP` to `NO_ELP_EXPECTED`.
